iceoryx_posh/include/iceoryx_posh/internal/popo/modern_api/base_subscriber.inl

```diff
@@ -104,46 +107,69 @@ inline cxx::expected<cxx::optional<Sample<const T>>, ChunkReceiveError> BaseSubs
     }
 }
 
-template <typename T, typename port_t>
-inline void BaseSubscriber<T, port_t>::releaseQueuedSamples() noexcept
+template <typename T, typename Subscriber, typename port_t>
+inline void BaseSubscriber<T, Subscriber, port_t>::releaseQueuedSamples() noexcept
 {
     m_port.releaseQueuedChunks();
 }
 
-template <typename T, typename port_t>
-inline void
-BaseSubscriber<T, port_t>::setConditionVariable(ConditionVariableData* const conditionVariableDataPtr) noexcept
+template <typename T, typename Subscriber, typename port_t>
+inline void BaseSubscriber<T, Subscriber, port_t>::invalidateTrigger(const uint64_t uniqueTriggerId) noexcept
 {
-    m_port.setConditionVariable(conditionVariableDataPtr);
+    if (m_trigger.getUniqueId() == uniqueTriggerId)
+    {
+        m_port.unsetConditionVariable();
+        m_trigger.invalidate();
+    }
 }
 
-template <typename T, typename port_t>
-inline void BaseSubscriber<T, port_t>::unsetConditionVariable() noexcept
+// ============================== Sample Deleter ============================== //
+
+template <typename T, typename Subscriber, typename port_t>
+inline BaseSubscriber<T, Subscriber, port_t>::SubscriberSampleDeleter::SubscriberSampleDeleter(port_t& port)
+    : m_port(std::ref(port))
 {
-    m_port.unsetConditionVariable();
 }
 
-template <typename T, typename port_t>
-inline bool BaseSubscriber<T, port_t>::hasTriggered() const noexcept
+template <typename T, typename Subscriber, typename port_t>
+inline void BaseSubscriber<T, Subscriber, port_t>::SubscriberSampleDeleter::operator()(T* const ptr) const
 {
-    return m_port.hasNewChunks();
+    auto header = mepoo::convertPayloadPointerToChunkHeader(reinterpret_cast<void*>(ptr));
+    m_port.get().releaseChunk(header);
 }
 
-// ============================== Sample Deleter ============================== //
-
-template <typename T, typename port_t>
-inline BaseSubscriber<T, port_t>::SubscriberSampleDeleter::SubscriberSampleDeleter(port_t& port)
-    : m_port(std::ref(port))
+template <typename T, typename Subscriber, typename port_t>
+inline cxx::expected<WaitSetError>
+BaseSubscriber<T, Subscriber, port_t>::attachTo(WaitSet& waitset,
+                                                [[gnu::unused]] const SubscriberEvent subscriberEvent,
+                                                const uint64_t triggerId,
```

Comment:
I still find it inappropriate to call it triggerId when leaving it to the caller to set it. Consider triggerType or similar. If it really was an id mechanism (which it is not), I would generate those internally.

Reply:
It is a id mechanism where to user can explicitly generate the ids on their end. The use case and a detailed description is provided in the `ice_waitset_grouping.cpp` example. 

This id is used to group multiple triggers together for instance that every subscriber has the same id since they are all handled in the same way by the user.

@MatthiasKillat do you have a name suggestion? Maybe just `identifier`, I know id is the abbreviation of identifier but maybe id sounds somehow unique...

Reply:
I thought about triggerType but am not happy with it. Would not change it for that. I just think it may be confusing for the novice user. Conceptually it is fine (i.e. we need some kind of value/type for this). The name may change later but the functionality will not.